shared/actions/fs-platform-specific.desktop.js

```diff
@@ -0,0 +1,101 @@
+// @flow
+import * as FsGen from './fs-gen'
+import * as Saga from '../util/saga'
+import fs from 'fs'
+import * as Constants from '../constants/config'
+import type {TypedState} from '../constants/reducer'
+import {shell} from 'electron'
+import {isLinux, isWindows} from '../constants/platform'
+import {navigateTo, switchTo} from './route-tree'
+import {fsTab} from '../constants/tabs'
+import logger from '../logger'
+
+// pathToURL takes path and converts to (file://) url.
+// See https://github.com/sindresorhus/file-url
+function pathToURL(path: string): string {
+  let goodPath = path.replace(/\\/g, '/')
+
+  // Windows drive letter must be prefixed with a slash
+  if (goodPath[0] !== '/') {
+    goodPath = '/' + goodPath
+  }
+
+  return encodeURI('file://' + goodPath).replace(/#/g, '%23')
+}
+
+function openInDefaultDirectory(openPath: string): Promise<*> {
+  return new Promise((resolve, reject) => {
+    // Paths in directories might be symlinks, so resolve using
+    // realpath.
+    // For example /keybase/private/gabrielh,chris gets redirected to
+    // /keybase/private/chris,gabrielh.
+    fs.realpath(openPath, (err, resolvedPath) => {
+      if (err) {
+        reject(new Error(`No realpath for ${openPath}: ${err}`))
+        return
+      }
+      // Convert to URL for openExternal call.
+      // We use openExternal instead of openItem because it
+      // correctly focuses' the Finder, and also uses a newer
```

Comment:
I know this is from the original kbfs saga, but I don't quite understand this part. When I experimented with `shell.openExternal`, it doesn't focus items in Finder, but would just open it (like when you use the `open` command).

Reply:
You're right, it doesn't seem to function according to the comment, at least not on Linux. I'll test macOS and Windows as well.

Reply:
Yeah, I think the showItemInFolder is the only API that does it: https://github.com/electron/electron/blob/master/docs/api/shell.md